### ISSUE TITLE
Group selection not possible

### ISSUE THREAD

**Issue by leofuehringer** [stjosh/auto_groups#55]
Hello,
we use Nextcloud 20 and Auto Groups 1.3. 
I try to select an "Auto Group". When I click at it - nothing happens. When I type text in the field, no group suggestion appears.
It's not possible to save any changes...
The following error appears in the console view:
```
Uncaught TypeError: Cannot read property 'setupGroupsSelect' of undefined
    at HTMLDocument.<anonymous> (admin.js?v=c92f5d6c-11:21)
    at l (jquery.js:3187)
    at Object.add [as done] (jquery.js:3246)
    at e.fn.init.m.fn.ready (jquery.js:3496)
    at admin.js?v=c92f5d6c-11:15
(anonymous) @ admin.js?v=c92f5d6c-11:21
l @ jquery.js:3187
add @ jquery.js:3246
m.fn.ready @ jquery.js:3496
(anonymous) @ admin.js?v=c92f5d6c-11:15
```

Does anyone have a hint?
Tanks, Leo

**Comment by stjosh** [comment]
Hi Leo, thanks for your report! Unfortunately, I can't reproduce this behavior (I have tested on NC 20.0.4). That makes it hard to fix 😉 . Do you happen to have any further inputs in order to reproduce?

**Comment by leofuehringer** [comment]
Hi, I'm not sure... Perhaps, I've added a group to "Auto Group" and after that I deleted this group...
If you like, I'll give you access to the nextcloud.

**Comment by stjosh** [comment]
I am really sorry, but I can‘t make the time to debug the issue on your system.

AutoGroups should actually prevent you from deleting groups that are in use by the app, so it is most likely something different (or you found a way to circumvent the deletion protection, but that‘s rather unlikely).

A few hints for further investigations:
- You might want to try uninstalling and re-installing the app. Does the problem persist?
- What browser are you using? Is the behavior different for different browsers?
- Pls report back on my question regarding the user/groups backend you are using - is it Nextcloud‘s default one?
- What‘s your exact Nextcloud version? (NC 20, but which sub-version?)
- Are you using other Apps which render their setting in the same settings page? Maybe a bug in another app‘s code is killing JavaScript execution and the JS for the AutoGroups app is never executed?

**Comment by seppultra** [comment]
Hi,
just i case someone is searching for this: Today i faced the same issue with Chrome (Version 90.0.4430.72). I switched to the latest Firefox and it works there. Latest Edge does work properly too.

I'm using NC 20.0.8 and Auto Groups 1.3.0.

**Comment by stjosh** [comment]
Many thanks for the hint @seppultra - I can confirm that I am able to reproduce the issue in Chrome as well. I'll dig deeper into this as soon as I find the time for it. 

Meanwhile, as a workaround, please use Firefox or Edge for editing the AutoGroups. Since this is a frontend issue, I expect the app itself to work correctly with all Browser for "normal" users, so it's only the admin page for changing the group configuration which is affected by this bug.

**Comment by stjosh** [comment]
I have to correct myself: It's not Browser dependent. After a few reloads, the group selection was also possible in Chrome. On the other hand, I can sporadically reproduce the issue in Firefox.

I think it has to do with a race condition - the code here might be executed before the object OC.Settings has been initialized:

https://github.com/stjosh/auto_groups/blob/master/js/admin.js#L21

I will add a timeout of 0 here, just to make sure this is delayed in the execution queue until all other document.ready handlers have been executed.

**Comment by stjosh** [comment]
I have just published release 1.3.1 which should fix this issue. Many thanks for reporting.
